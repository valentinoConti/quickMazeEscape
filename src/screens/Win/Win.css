#win {
  width: 100%;
  height: 100%;
  text-align: center;
}

#win > p {
  margin-top: 20px;
  font-size: 36px;
}

#win > p:first-child {
  margin-top: 100px;
  font-size: 72px;
  font-weight: bold;
}

#win > .menuButton {
  margin-top: 200px;
  width: 300px;
  height: 50px;
  margin-left: 0;
}