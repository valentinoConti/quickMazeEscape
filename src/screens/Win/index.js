import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import './Win.css';

const Win = ({setScreen, score, setScore}) => {
  const goMenu = () => {
    setScore(0);
    setScreen('menu');
  }

  const highScore = JSON.parse(localStorage.getItem('maze-game-highscore'));
  if (!highScore || highScore < score) {
    localStorage.setItem('maze-game-highscore', score);
  }

  return (
    <div id="win">
      <p>You win!</p>
      <p>Your score: <b id="actualScore">{score}</b></p>
      <p>Highest score: <b id="highestScore">{highScore < score ? score : highScore}</b></p>
      <button className="menuButton" onClick={goMenu}>
        Back to menu
      </button>
    </div>
  )
};

Win.propTypes = {
  setScreen: PropTypes.func.isRequired,
  score: PropTypes.number.isRequired
};

export default Win;
