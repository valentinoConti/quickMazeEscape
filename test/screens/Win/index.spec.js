import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { configure, mount } from 'enzyme';
import Win from '../../../src/screens/Win';

configure({adapter: new Adapter()});

describe('Menu screen test', () => {
  jest.spyOn(Object.getPrototypeOf(window.localStorage), 'getItem');
  const mockSetScreen = jest.fn();
  const mockSetScore = jest.fn();
  const mockScore = 20;

  it('should take/compare snapshot', () => {
    const wrapper = mount(
      <Win
        score={mockScore}
        setScore={mockSetScore}
        setScreen={mockSetScreen}
      />
    );

    expect(wrapper).toMatchSnapshot();
  })

  it('should test back to Menu button with new better score', () => {
    const wrapper = mount(
      <Win
        score={mockScore}
        setScore={mockSetScore}
        setScreen={mockSetScreen}
      />
    );

    expect(localStorage.getItem).toHaveBeenCalled();
    expect(wrapper.find('#win')).toBeTruthy();
    expect(mockSetScreen).not.toHaveBeenCalled();
    expect(wrapper.find('#actualScore').text()).toBe('20');
    expect(wrapper.find('#highestScore').text()).toBe('20');
    wrapper.find('button').simulate('click');
    expect(mockSetScreen).toHaveBeenCalled();
    expect(mockSetScore).toHaveBeenCalled();

    wrapper.unmount();
  });

  it('should test previous higher score', () => {
    Object.getPrototypeOf(window.localStorage).getItem = jest.fn(() => 45);

    const wrapper = mount(
      <Win
        score={mockScore}
        setScore={mockSetScore}
        setScreen={mockSetScreen}
      />
    );

    expect(localStorage.getItem).toHaveBeenCalled();
    expect(wrapper.find('#win')).toBeTruthy();
    expect(wrapper.find('#actualScore').text()).toBe('20');
    expect(wrapper.find('#highestScore').text()).toBe('45');
    expect(mockSetScore).toHaveBeenCalled();
  });
});