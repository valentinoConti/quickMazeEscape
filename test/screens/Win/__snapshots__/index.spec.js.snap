// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu screen test should take/compare snapshot 1`] = `
<Win
  score={20}
  setScore={[MockFunction]}
  setScreen={[MockFunction]}
>
  <div
    id="win"
  >
    <p>
      You win!
    </p>
    <p>
      Your score: 
      <b
        id="actualScore"
      >
        20
      </b>
    </p>
    <p>
      Highest score: 
      <b
        id="highestScore"
      >
        20
      </b>
    </p>
    <button
      className="menuButton"
      onClick={[Function]}
    >
      Back to menu
    </button>
  </div>
</Win>
`;
